File: cs470/proj1/Tennis/Match.cpp
// Changes made by Zachary Thompson

#include "Match.hpp"
#include "MatchScore.hpp"
#include "Set.hpp"
#include "Player.hpp"

Match::Match( Player *p1, Player *p2 ): Competition( p1, p2 ) {}
Score *Match::play( Player *p ) {

    Score* score = new MatchScore( player1(), player2() );
    int setCount = 0;

    while ( !score->haveAWinner() ) {
        Set* set = new Set( player1(), player2() );
        Player* server = ++setCount % 2 == 1 ? p : Player::otherPlayer(p);
        ((MatchScore*) score) -> addScore( set -> play( server  ) );
        delete set;
    }

    return score;
}

File: cs470/proj1/ObjectiveCTennis/TennisInObjectiveC/Match.m
//  Additions made by Zachary Thompson
//  Match.m
//  TennisInObjectiveC
//
//  Created by student on 2/3/14.
//  Copyright (c) 2014 Ali Kooshesh. All rights reserved.
//

#import "Match.h"
#import "MatchScore.h"
#import "Set.h"

@implementation Match

-(Score *) play:(Player *)player
{
    Score *matchScore = [[MatchScore alloc] initWithFirstPlayer:self.player1 secondPlayer:self.player2];
    int setCount = 0;
    
    while( ! [matchScore haveAWinner]) {
        Player *server = setCount++ % 2 == 0 ? player :[Player otherPlayer:player];
        Set* set = [[Set alloc] initWithFirstPlayer:self.player1 secondPlayer:self.player2];
        [(MatchScore*)matchScore addSetScore:[set play:server]];
        set = nil;
    }
    return matchScore;
}

@end
